### PR TITLE
DataTable split

### PR DESCRIPTION
**Pull request by TorsteinHonsi**
A separate PR for a simplified DataTable base, required for reducing the file size when moving to [DataTable-powered data handling](https://github.com/highcharts/highcharts/pull/20530) in Highcharts Core.

### To do
 - [x] Fix `setColumns` now probably triggers the `afterSetColumns` event twice.
 - [x] Fix `setRow` lacking the `afterSetRows` event.

### Discussion points
* Q: Should `alias` support be removed all over? _A: maybe, but let the Dash team decide_.
* Q: Do we need the before events at all, or are they just for compliance with the before/after convention? These are currently unused: `cloneTable`, `deleteColumns`, `deleteRows`, `setCell`, `setColumns`, `setRows`. Also, they don't offer the possibility to prevent default, like the convention is in Highcharts. _A: Decide after the `DataView` is implemented. It will overturn modifiers and all of this may be obsolete._
* Q: The modifier calls (`tableModifier.modifyColumns`) etc - can we move those to event handlers in the modifier? And a bigger question - is it a limitation that each table only has one modifier and `modified`? _A: As above_
* Q: Is it better to use one `DataTable` class but have a composition for the full-featured stuff? Otherwise we are always loading DataTableCore for the core Series etc. And the Highcharts object will expose both `DataTable` and `DataTableCore`. _A: Keep it as inheritance. Modules can load/check their own required class, or in worst case cast using `setPrototypeOf`._

### CODE REVIEW

File: ts/tsconfig.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
         // "lib": [],                                        /* Specify a set of bundled library declaration files that describe the target runtime environment. */
         "lib": [
             "DOM",
-            "ES2016"
+            "ES2017"
```
Comment (bre1470) [inline ts/tsconfig.json:20]: We need IE11 polyfill for `Object.entries` and `Object.values`, right?

Also it is better to keep `ES2016` and add `ES2017.Object`, to keep the amount of required polyfills in check.

**Review by bre1470 — CHANGES_REQUESTED** [review]
Looks promising!

File: ts/tsconfig.json (hunk 1 above)
Comment (TorsteinHonsi) [reply ts/tsconfig.json:20]: I'll redo this, I used `Object.values` temporarily, but undid it.

Comment (TorsteinHonsi) [reply ts/tsconfig.json:20]: Looks like the polyfills are already there, so I'll just add `ES2027.Object`.

https://github.com/highcharts/highcharts/blob/234497b961fede9457e1943e0c4529fe144aefc7/ts/masters-es5/highcharts.src.ts#L48-L73

**Comment by TorsteinHonsi** [comment]
> As for aliases, we discussed it and unanimously agreed that we don't need it, so this class can be simplified much more. :)

@vazonik Okay, good! I'll leave it out of this PR though. I already removed aliases from DataTableCore.

File: ts/Data/DataTableCore.ts [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+    /**
+     * Constructs an instance of the DataTable class.
+     *
+     * @param {Highcharts.DataTableOptions} [options]
+     * Options to initialize the new DataTable instance.
+     */
+    public constructor(
+        options: DataTableOptions = {}
+    ) {
+        this.columns = {};
```
Comment (bre1470) [inline ts/Data/DataTableCore.ts:67]: Double `this.columns`.

File: ts/Data/DataTableCore.ts [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+
+        /**
+         * Whether the ID was automatic generated or given in the constructor.
+         *
+         * @name Highcharts.DataTable#autoId
+         * @type {boolean}
+         */
+        this.autoId = !options.id;
+        this.columns = {};
```
Comment (bre1470) [inline ts/Data/DataTableCore.ts:76]: Double `this.columns`.

**Review by bre1470 — APPROVED** [review]
One thing to improve. Otherwise this looks good! 👍

**Pull request merged**